### PR TITLE
Stop FTEs from registering for intern events

### PR DESCRIPTION
**Pull request by digit01Wave**
#22 

Fixes a part of the above bug by stopping FTEs from registering for intern events

### CODE REVIEW

File: ServerCore/Helpers/TeamHelper.cs [hunk 1]
```diff
@@ -136,5 +136,14 @@ public static async Task DeleteTeamAsync(PuzzleServerContext context, Team team)
 
             return new Tuple<bool, string>(true, "");
         }
+
+        /// <summary>
+        /// Returns whether or not a user is a full-time microsoft employee
+        /// </summary>
+        /// <returns>True if the user is a full-time microsoft employee and false otherwise</returns>
+        public static bool IsFTE(string email)
```
Comment (tabascq) [inline ServerCore/Helpers/TeamHelper.cs]: Maybe IsNonIntern, since contractors are not FTEs...

Comment (tabascq) [reply ServerCore/Helpers/TeamHelper.cs]: (though contractors are not eligible either so the logic is fine - the name and comment are slightly inaccurate though)

File: ServerCore/Pages/Teams/Apply.cshtml.cs [hunk 2]
```diff
@@ -37,6 +37,11 @@ public async Task<IActionResult> OnGet(int teamID, string password)
                 return Forbid();
             }
 
+            if (Event.IsInternEvent && TeamHelper.IsFTE(LoggedInUser.Email))
+            {
+                return NotFound("FTEs cannot sign up for this intern-only event.");
```
Comment (tabascq) [inline ServerCore/Pages/Teams/Apply.cshtml.cs]: This message should say something like "Microsoft employees and contractors"

File: ServerCore/Pages/Teams/List.cshtml [hunk 3]
```diff
@@ -6,10 +6,19 @@
...
+    Boolean canRegister = Model.EventRole == ModelBases.EventRole.play && Model.Event.IsTeamMembershipChangeActive && !Model.IsFTEInInternEvent;
 }
 
 <p>
-    @if (DateTime.UtcNow < Model.Event.TeamRegistrationBegin)
+    @if (Model.IsFTEInInternEvent)
+    {
+        <h2>Registration Not Allowed for @Model.Event.Name</h2>
+        <div>
+            We are sorry. This is an intern-only event, so no FTEs can register.
```
Comment (tabascq) [inline ServerCore/Pages/Teams/List.cshtml]: More FTEness

File: ServerCore/Pages/Teams/Apply.cshtml.cs (hunk 2 above)
Comment (digit01Wave) [reply ServerCore/Pages/Teams/Apply.cshtml.cs]: I realize that in this case, forbidden access made more sense.

File: ServerCore/Pages/Teams/Create.cshtml.cs [hunk 4]
```diff
@@ -36,14 +37,14 @@ public async Task<IActionResult> OnGetAsync()
...
 
-            if (EventRole == EventRole.admin && !await LoggedInUser.IsAdminForEvent(_context, Event))
+            if (!Event.IsTeamRegistrationActive && EventRole != EventRole.admin)
             {
-                return Forbid();
+                return NotFound();
             }
 
-            if (!Event.IsTeamRegistrationActive && EventRole != EventRole.admin)
+            if (Event.IsInternEvent && TeamHelper.IsMicrosoftNonIntern(LoggedInUser.Email) && EventRole != EventRole.admin)
```
Comment (tabascq) [inline ServerCore/Pages/Teams/Create.cshtml.cs]: IsNotAllowedInInternEvent() check above handles this case, yes?

File: ServerCore/Pages/Teams/Create.cshtml.cs [hunk 5]
```diff
@@ -26,7 +27,9 @@ public async Task<IActionResult> OnGetAsync()
                 return Challenge();
             }
 
-            if (EventRole != EventRole.play && EventRole != EventRole.admin)
+            if ((EventRole != EventRole.play && EventRole != EventRole.admin)
+                || IsNotAllowedInInternEvent()
+                || EventRole == EventRole.admin && !await LoggedInUser.IsAdminForEvent(_context, Event))
```
Comment (morganbr) [inline ServerCore/Pages/Teams/Create.cshtml.cs]: Does this line needs parens to be correct?

**Pull request merged**